### DISCUSSION TITLE
Unique constraint on multiple columns

### DISCUSSION THREAD

**Discussion by penberg** [SeaQL/sea-query#180] [Q&A] [answered]
SQL allows defining unique constraint for a table on multiple columns. For example, with `sqlite` the syntax looks as follows:

```sql
sqlite> CREATE TABLE person (
    first_name varchar(255) NOT NULL,
    last_name varchar(255) NOT NULL,
    UNIQUE (first_name, last_name)
);
```

However, there is no way to express this with the `sea-query` API. There is a `unique_key()` function, but that is limited to a single column.

**Comment by tyt2y3** [comment, marked as answer]
It should be possible to do something like this

```rust
let index = Index::create()
    .name("idx-glyph-aspect")
    .table(Glyph::Table)
    .col(Glyph::Aspect)
    .col(Glyph::Ratio)
    .unique();
```

let me know if it works!

**Comment by penberg** [comment]
@tyt2y3 Thanks, that works! It would be nice to be able to specify the constraint without the index as well, though.

**Comment by Sytten** [comment]
FYI unique constraint are implemented with a unique index. It is just a syntax sugar.
